clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -544,10 +545,14 @@ public void handle(JoinGroupResponse joinResponse, RequestFuture<ByteBuffer> fut
                 future.raise(error);
             } else if (error == Errors.INCONSISTENT_GROUP_PROTOCOL
                     || error == Errors.INVALID_SESSION_TIMEOUT
-                    || error == Errors.INVALID_GROUP_ID) {
-                // log the error and re-throw the exception
+                    || error == Errors.INVALID_GROUP_ID
+                    || error == Errors.GROUP_MAX_SIZE_REACHED) {
                 log.error("Attempt to join group failed due to fatal error: {}", error.message());
-                future.raise(error);
+                if (error == Errors.GROUP_MAX_SIZE_REACHED) {
```

Comment:
I think it would be good to have the log
```
log.error("Attempt to join group failed due to fatal error: {}", error.message());
```

Maybe we could move the `GROUP_AUTHORIZATION` to this level?